community/kernel/src/main/java/org/neo4j/kernel/api/index/util/FailureStorage.java

```diff
@@ -123,6 +124,18 @@ public synchronized void storeIndexFailure( long indexId, String failure ) throw
             channel.force( true );
             channel.close();
         }
+        catch ( FileNotFoundException fnf )
```

Comment:
That, or add a comment explaining the reason this catch exists. Either works i guess.